### PR TITLE
Refactor: func `Save` in package `pkg/cmd/image` shouldn't create file from `options.Output`

### PR DESCRIPTION
**Pull request by rkonfj**
> `pkg` side should assume a `StdOut` is given (i.e., not create a file from `options.Output`)

Signed-off-by: Yakun Jing <rkonfj@gmail.com>

### CODE REVIEW

**Comment by rkonfj** [comment]
What I'm a bit unsure about is whether we need to move the following code into the `main` package of command?
https://github.com/containerd/nerdctl/blob/bba23d57744dbdd14c7674efab3e743600a3ba1f/pkg/cmd/image/save.go#L35-L44

**Comment by djdongjin** [comment]
> What I'm a bit unsure about is whether we need to move the following code into the main package of command?

Yes this should be part of the `cmd` side (i.e., cmd/nerdctl/image_save.go). `pkg` side should assume a `StdOut` is given (i.e., not create a file from `options.Output`)

File: cmd/nerdctl/image_save.go [hunk 1]
```diff
@@ -69,29 +67,56 @@ func processImageSaveOptions(cmd *cobra.Command) (types.ImageSaveOptions, error)
...
 		AllPlatforms: allPlatforms,
-		Output:       output,
 		Platform:     platform,
 	}, err
 }
 
-func saveAction(cmd *cobra.Command, args []string) error {
-	if len(args) == 0 {
-		return fmt.Errorf("requires at least 1 argument")
+func determineSaveToFile(cmd *cobra.Command) (*os.File, error) {
```
Comment (suyanhanx) [inline cmd/nerdctl/image_save.go:75]: Maybe `determineSaveToFile` -> `determineFileToSave` ?

File: cmd/nerdctl/image_save.go [hunk 2]
```diff
@@ -69,29 +67,56 @@ func processImageSaveOptions(cmd *cobra.Command) (types.ImageSaveOptions, error)
...
 		AllPlatforms: allPlatforms,
-		Output:       output,
 		Platform:     platform,
 	}, err
 }
 
-func saveAction(cmd *cobra.Command, args []string) error {
-	if len(args) == 0 {
-		return fmt.Errorf("requires at least 1 argument")
+func determineFileToSave(cmd *cobra.Command) (*os.File, error) {
```
Comment (djdongjin) [inline cmd/nerdctl/image_save.go:75]: maybe just put the logic within `saveAction`, which should be cleaner. Most of the logic are just err check.

**Review by djdongjin — APPROVED** [review]

**Review by suyanhanx — APPROVED** [review]

**Comment by suyanhanx** [comment]
close #1940 .

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**